### PR TITLE
Add RebootWorker to geomaster data provider

### PR DESCRIPTION
**Pull request by michimune**
This PR adds a new API to the geomaster data provider named RebootWorker. This allows detectors to reboot a worker in the specified server farm.

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 1]
```diff
@@ -489,6 +489,39 @@ public Task<string> GetLinuxContainerLogs(string subscriptionId, string resource
...
+        ///     var serverFarmName = (string)observerData.server_farm_name;
+        ///     var machineName = "RD0123456789";
+        ///
+        ///     var results = await dp.GeoMaster.RebootWorker(subId, rg, serverFarmName, machineName);
+        /// }
+        /// </code>
+        /// </example>
+        public Task<string> RebootWorker(string subscriptionId, string resourceGroup, string serverFarmName, string workerName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var path = $"subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Web/serverFarms/{serverFarmName}/workers/{workerName}/reboot";
```
Comment (ShekharGupta1988) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:521]: nitpick : can you add basic checks for the arguments?

Comment (michimune) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:521]: You are right. I will add null checking for the parameters.

**Comment by michimune** [comment]
That's my concern, too. Do you have a good idea to control?

**Comment by ShekharGupta1988** [comment]
> That's my concern, too. Do you have a good idea to control?

Lets discuss this. The real question is do we want to do this via Diag Backend --> Geomaster  (since this will open a can of worms) or should we let "users" issue this reboot via ARM --> Geomaster  (much safer, they can only reboot their own workers etc)

**Comment by michimune** [comment]
The latter approach is the best. I believe we are already doing it with some features, correct? We have some diagnostic tools which are only allowed to access caller's resources

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 1 above)
Comment (michimune) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs:541]: Added null checks

**Comment by ShekharGupta1988** [comment]
> The latter approach is the best. I believe we are already doing it with some features, correct? We have some diagnostic tools which are only allowed to access caller's resources

correct. Basically if you are looking to provide a way to add this "Reboot worker" option in detectors, then we should look to add this as a solution. When this gets rendered :
- Internally in applens, it will have customer ready content that CSS can share with the customers.
- Externally in Case Submission(or Diag and Solve), customers will be able to reboot their workers via ARM call (and OAuth token)

**Comment by michimune** [comment]
Got it, thank you. I will update my changes.

**Comment by michimune** [comment]
It turned out that we don't need this PR at all. It is going to be replaced with this PR instead: https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/pull/854

**Pull request closed without merging**